### PR TITLE
Create classes to store the response map. 

### PR DESCRIPTION
**Pull request by StrangeNoob**
**What kind of change does this PR introduce?**
- Refactor the User Model.
- Extract Common Class i.e., Joined Organization in Switch_org.dart and User,dart  
- Implement the User Model where Required

**Did you add tests for your changes?**
Yes. I have made some Changes
**If relevant, did you update the documentation?**
No.
**Summary**

**Issue**:  #720 
**Does this PR introduce a breaking change?**
No. 

**Other information**
This PR is now complete I have tried converted as much as response to the models so that we can achieve MVVM architecture easily.

### CODE REVIEW

**Comment by StrangeNoob** [comment]
Hello @Sagar2366 @palisadoes @dellyjm @DeltaHarbinger,
Can you just check If I am on the right path or did I need to change something? 
If its right then I will be able to continue to implement in Newsfeeds, Events, Member Page Also.

**Comment by palisadoes** [comment]
@Sagar2366 @dellyjm @DeltaHarbinger Please take a look

File: lib/model/joinedorganization.dart [hunk 1]
```diff
@@ -0,0 +1,146 @@
+class JoinedOrganization {
+  JoinedOrganization({
```
Comment (sumitra19jha) [inline lib/model/joinedorganization.dart:2]: @StrangeNoob If possible can you add the `required` condition for some of the parameter like id

Comment (StrangeNoob) [reply lib/model/joinedorganization.dart:2]: Sure No problem will look into more details

**Review by DeltaHarbinger — APPROVED** [review]
Using this User model is better than just having maps

**Comment by DeltaHarbinger** [comment]
@StrangeNoob, this pull request seems fine. We should be able to merge it

**Comment by StrangeNoob** [comment]
@palisadoes @DeltaHarbinger @Sagar2366  I made required the changes as suggested by @sumitra19jha. Can you please review it.

**Comment by TheHazeEffect** [comment]
Hey @StrangeNoob Nice this is pretty impressive.

@DeltaHarbinger @sumitra19jha, I'm thinking it may be best we use the [Json_Serilizable](https://pub.dev/packages/json_serializable) package to automate Json conversion, less room for error and faster development, we would also update the readme to mention running the scaffold command for project setup

**Comment by sumitra19jha** [comment]
> Hey @StrangeNoob Nice this is pretty impressive.
> 
> @DeltaHarbinger @sumitra19jha, I'm thinking it may be best we use the [Json_Serilizable](https://pub.dev/packages/json_serializable) package to automate Json conversion, less room for error and faster development, we would also update the readme to mention running the scaffold command for project setup

It can be done and seems like a good idea @TheHazeEffect

**Comment by StrangeNoob** [comment]
Hey @TheHazeEffect and @sumitra19jha,
Hope you are doing well in this terrible time.
I having using the response creating a class out of it with the [quicktype.io](https://app.quicktype.io/). So I haven't looked into the  [Json_Serilizable](https://pub.dev/packages/json_serializable). I wont mind changing those files to [Json_serilizable](https://pub.dev/packages/json_serializable) models. I don't think it will  make any difference to the models @TheHazeEffect? Can you explain it to me.

**Comment by TheHazeEffect** [comment]
> Hey @TheHazeEffect and @sumitra19jha,
> Hope you are doing well in this terrible time.
> I having using the response creating a class out of it with the [quicktype.io](https://app.quicktype.io/). So I haven't looked into the [Json_Serilizable](https://pub.dev/packages/json_serializable). I wont mind changing those files to [Json_serilizable](https://pub.dev/packages/json_serializable) models. I don't think it will make any difference to the models @TheHazeEffect? Can you explain it to me.

Hi,
I believe this PR achieves the original intent, refactoring to use Json_Serializable can be done as a separate PR so I'm onboard with this being merged.  For now you can just work on resolving the merge conflicts

As for differences, not much, it just automates building out the json mapping and I guess also reduces the number of lines in your model classes

**Comment by StrangeNoob** [comment]
> I believe this PR achieves the original intent, refactoring to use Json_Serializable can be done as a separate PR so I'm onboard with this being merged. For now you can just work on resolving the merge conflicts
> As for differences, not much, it just automates building out the json mapping and I guess also reduces the number of lines in your model classes

Sure No Problem. I would look to work on that too. Thanks, @TheHazeEffect for clarifying. I have resolved the conflicts and test them. Can you please review it?

**Comment by palisadoes** [comment]
@dellyjm
@DeltaHarbinger
@Sagar2366
@TheHazeEffect

Please review this

**Review by Sagar2366 — APPROVED** [review]

**Comment by palisadoes** [comment]
@StrangeNoob please resolve the conflicts

**Comment by StrangeNoob** [comment]
@palisadoes It's done. Can you please review it?

**Comment by palisadoes** [comment]
> @palisadoes It's done. Can you please review it?

@StrangeNoob File `lib/views/pages/newsfeed/news_article.dart` has a conflict

**Comment by palisadoes** [comment]
@StrangeNoob still having conflicts.

**Comment by StrangeNoob** [comment]
@palisadoes It's happening because I have replaced major Maps with objects and classes. There are changes in every file possible. You can merge this now

**Pull request merged**